src/index.js

```diff
@@ -1,117 +1,164 @@
 import dns from 'dns';
 import setConfig from './config';
-import Context from './context';
-import Callback from './callback';
 import Report from './report';
 import globals from './globals';
 
-function makeDnsPromise(host) {
+const dnsPromises = {};
+
+function getDnsPromise(host) {
+  if (dnsPromises[host]) {
```

Comment:
Yeah, we shouldn't be caching the result... and the system itself DOES do caching. For most invocations this will call a C function that fetches the DNS from a cache anyway. _Sometimes_ the DNS will be expired and it will do a new query.

I thought we would do `dnsPromises[host] = new Promise()` instead